state/GlobalStoreProvider/GlobalStoreProvider.tsx

```diff
@@ -25,14 +30,29 @@ const GlobalStoreProvider = ({ children }: PropsWithChildren) => {
 
   const [signUpPhone, setSignUpPhone] = useState<string | null>(null)
 
+  const [user, setUser] = useState<CognitoUser | null>(null)
+
+  const mmkvStorage = useMemo(() => {
+    if (user) {
+      const username = user.getUsername()
+
+      return new MMKV({ id: `user-${username}-storage` })
+    }
+
+    return new MMKV()
+  }, [user])
```

Comment:
I think we do not need to put storage into context. Instead, this hooks could be used:
https://github.com/mrousavy/react-native-mmkv/blob/master/docs/HOOKS.md#reactively-use-individual-keys-from-custom-instances

```
  const userStorage = useMMKV({ id: `${userId}.storage` })
```

Reply:
very good, I will change it